specification/src/chapter3.adoc

```diff
@@ -1,45 +1,313 @@
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| PMP/ePMP MUST be used to protect machine mode from any supervisor domain.
+
+| CAT_NN
+| Either PMP/ePMP or MTT MUST be used to enforce physical memory isolation boundaries for supervisor domains.
+
+|===
+
+PMP can be used for more static and deterministic use cases.
+
+MTT can be used where more fine grained dynamic resource management across supervisor domain boundaries is required.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+
+| ID#     
+| Requirement
+
+| CAT_NNN
+| A system supporting supervisor domains MUST support supervisor domain extensions for interrupts and performance counters, and SHOULD support supervisor domain extensions for external debug.
+
+|===
+
+Interrupts: https://github.com/riscv/riscv-smmtt +
+External debug: https://github.com/riscv-non-isa/riscv-external-debug-security +
+Performance counters: 
+
+These extensions enable management of interrupts, external debug, and performance counters across supervisor domain boundaries:
+
+* External debug can be enabled for one supervisor domain without affecting other supervisor domains
+* M-mode firmware manage interrupt routing and preemption across supervisor domain boundaries
+* M-mode firmware can ensure that performance counters cannot be used by software in one supervisor domain to measure operations in other supervisor domains 
+
+==== MTT
+
+https://github.com/riscv/riscv-smmtt
+
+The _memory tracking table (MTT)_ is a memory structure managed by machine mode, tracking memory ownership across supervisor domains. It is designed to enable fine grained dynamic memory management across supervisor domain boundaries, with policy typically set by a hypervisor in a hosting domain responsible for resource management.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| PMP/ePMP MUST be used to protect M-mode from lower privilege levels
+
+| CAT_NNN  
+| MTT configurations MUST only be directly accessible to machine mode
+|===
+
+==== IOPMP
+
+https://github.com/riscv-non-isa/iopmp-spec
+
+IOPMP is a system level component providing physical memory access control for device initiated transactions, complementing PMP and sPMP rules. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| A system which supports PMP/ePMP, or sPMP, MUST implement IOPMP for device access control.
+
+| CAT_NNN
+| IOPMP configurations MUST only be directly accessible to machine mode.
+|===
+
+
+==== IOMTT
+
+https://github.com/riscv/riscv-smmtt
+
+IOMTT is a system level component providing physical memory access control for device initiated transactions, complementing MTT rules.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| A system which supports MTT MUST implement IOMTT for device initiated access control.
```

Comment:
Can IOMTT be implemented without an IOMMU? I think both are required.

Reply:
IOMMU is not required for IOMTT. The IOMMU may be disabled or not-implemented. The IOMTT are required to protect against implicit accesses from IOMMU itself as the IOMMU are controlled by the supervisor domains - similar to how MMU is controlled by the SD.

Reply:
per discussion above - leaving requirement as written